### PR TITLE
[Helm] Automate datasource import

### PR DESCRIPTION
**Pull request by ejianu**
### SUMMARY
Helm chart change: Added 2 possible properties which will essentially automate the creation of database connections in superset.

Changes in the values file:
`extraConfigs` to which we can pass a yaml file that we exported previously, that defines a datasource.
`extraConfigMountPath` location at which this file will be

Changes in the chart:

Added a configMap template file that only creates the ConfigMap if the `extraConfigs` is set. 
In the init-job file, the new "extra-config" volume is mounted only if `extraConfigs` is set.

In this way, the original approach should not be affected.

### TEST PLAN
1. Make sure you create a datasource in the interface and export it to yaml, then delete it and check it was deleted.
2. In the values file set the yaml you exported, this is an example, but any datasource exported yaml should do:
```
extraConfigs: 
  datasources.yaml: | 
      databases:
      - allow_csv_upload: true
        allow_ctas: true
        allow_cvas: true
        database_name: presto
        extra: "{\r\n    \"metadata_params\": {},\r\n    \"engine_params\": {},\r\n    \"\
          metadata_cache_timeout\": {},\r\n    \"schemas_allowed_for_csv_upload\": []\r\n\
          }"
        sqlalchemy_uri: presto://presto.whatever
        tables: []
```

3. set the following property, the datasource file will be mounted at some location:
`extraConfigMountPath: "/app/configs"`

4. under the Init job configuration in the init script add a command that imports the file we defined previously:
    echo "Initializing connection.... "
    superset import_datasources -p {{ .Values.extraConfigMountPath }}/datasources.yaml

5. Check if the datasource was imported correctly.

### CODE REVIEW

File: helm/superset/templates/configmap-superset.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name: {{ template "superset.fullname" . }}-extra-config
+  labels:
+    app: {{ template "superset.name" . }}
+    chart: {{ template "superset.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+{{- range $path, $config := .Values.extraConfigs }}
+  {{ $path }}: |
+{{ $config | indent 4 -}}
```
Comment (craig-rueda) [inline helm/superset/templates/configmap-superset.yaml:30]: Can you pipe the val to tpl?

So, {{ tpl $config . | indent 4 -}}

**Review by craig-rueda — CHANGES_REQUESTED** [review]
One quick thing, otherwise LGTM

**Comment by ejianu** [comment]
Changes made :)

**Review by craig-rueda — APPROVED** [review]

**Comment by willbarrett** [comment]
Please trim trailing whitespace to fix the precommit hook, after that should be good to merge.

File: helm/superset/values.yaml [hunk 2]
```diff
@@ -238,4 +253,4 @@ nodeSelector: {}
 
 tolerations: []
 
-affinity: {}
+affinity: {}
```
Comment (nytai) [inline helm/superset/values.yaml:256]:
```suggestion
affinity: {}

```

File: helm/superset/templates/configmap-superset.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    app: {{ template "superset.name" . }}
+    chart: {{ template "superset.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+{{- range $path, $config := .Values.extraConfigs }}
+  {{ $path }}: |
+{{ tpl $config . | indent 4 -}}
+{{- end -}}
+{{- end -}}
```
Comment (nytai) [inline helm/superset/templates/configmap-superset.yaml:32]:
```suggestion
{{- end -}}

```

**Comment by andreas-eberle** [comment]
What is the status on this issue?

**Comment by zhmurko** [comment]
if someone is looking why import with `extraConfigs` does not work, just add into `initscript` a command:

```
        echo "Initializing connection.... "
        superset import_datasources -p {{ .Values.extraConfigMountPath }}/datasources-init.yaml
```
this step is not present in master branch at this moment.

**Comment by jawabuu** [comment]
@ejianu @craig-rueda I have noticed that databases imported in this manner **do not** mask the password value like the ones added in the UI. Is this expected?

**Pull request merged**